.github/workflows/tests.yaml

```diff
@@ -261,8 +258,38 @@ jobs:
           # build, so simply inject an explicit no-op test target.
           echo "//scripts:no_op_test" >> $TARGETS_FILE
 
-          bazelisk test -c ${{ matrix.build_mode }} \
-            --target_pattern_file=$TARGETS_FILE
+          for i in {1..10}; do
+            bazel_exit=0
+            bazelisk test -c ${{ matrix.build_mode }} \
+              --target_pattern_file=$TARGETS_FILE || bazel_exit=$?
+
+            # If we succeed, we're done.
+            if (( $bazel_exit == 0 )); then
+              break
+            fi
+
+            # Several error codes are reliably permanent, break immediately.
+            # `1`  -- The build failed.
+            # `2`  -- Command line or environment problem.
+            # `3`  -- Tests failed or timed out, we don't retry at this layer
+            #         on execution timeout.
+            # `4`  -- No tests found, which should be impossible here.
+            # `8`  -- Explicitly interrupted build.
+            # `36` -- A suspected permanent error in the local execution
+            #         environment.
```

Comment:
Comment doesn't match the code.
36 is the exit code I have observed on transient failures -- I think that is worth stating that explicitly since it is not what the documented exit codes say.

Reply:
Doh, good call on the comment.

But extra doh on the fact that this is actually the exit code we keep seeing. Good suggestion to document it, comment updated and code left alone.